src/bundles/helia.js

```diff
@@ -0,0 +1,213 @@
+      return Object.assign({}, state, { ipfsReady: false, error: error })
+    }
+
+    return state
+  },
+
+  getExtraArgs () {
+    return { getIpfs: () => globalThis._ipfs }
+  },
+
+  selectIpfsReady: state => state.ipfs.ipfsReady,
+
+  selectIpfsIdentity: state => state.ipfs.identity,
+
+  doInitIpfs: () => async ({ dispatch, getState }) => {
+    console.log('Looking for IPFS')
+    console.time('IPFS_INIT')
+    dispatch({ type: 'IPFS_INIT_STARTED' })
+
+    let ipfs = null
+    let identity = null
+
+    // TODO: use delegatedContent&Peer routing via helia instead of kubo-rpc-client directly.
+    // TRY kubo-rpc-client (kubo-rpc-client)
+    const apiOpts = Object.assign(
+      {},
+      getState().ipfs.apiOpts,
+      getUserOpts('ipfsApi')
+    )
+    // TRY helia!
+    try {
+      console.time('HELIA_INIT')
+      console.log('Trying to start in-page helia')
+      console.info(
+        "🎛️ Customise your kubo-rpc-client opts by setting an `ipfsApi` value in localStorage. e.g. localStorage.setItem('ipfsApi', JSON.stringify({port: '1337'}))"
+      )
+      ipfs = await initHelia(apiOpts)
+      console.log('got ipfs')
+      // identity = await ipfs.libp2p.identify()
+      identity = 'helia-TBD'
+      console.log('in-page helia instance ready!', identity)
+      globalThis._ipfs = ipfs
+      console.timeEnd('HELIA_INIT')
+      console.timeEnd('IPFS_INIT')
+      return dispatch({
+        type: 'IPFS_INIT_FINISHED',
+        payload: {
+          identity,
+          provider: 'helia'
+        }
+      })
+    } catch (error) {
+      if (error.message && error.message.includes('subtle is undefined')) {
+        console.warn('IPLD Explorer requires access to window.crypto, redirecting to canonical URL that is known to provide it in all browsers')
+        // This error means js-ipfs was loaded in a context that is not marked
+        // as Secure Context by the browser vendor.  (example: *.localhost in
+        // Firefox until https://bugzilla.mozilla.org/show_bug.cgi?id=1220810
+        // is addresssed)
+        // This is difficult to debug for regular user, as Explorer simply fails to load anything from IPFS.
+        // For now, we detect this failure and redirect to canonical version with TLS, so it always works.
+        const url = new URL('https://explore.ipld.io/?x-ipfs-companion-no-redirect')
+        // url.hash = window.location.hash
+        // window.location.replace(url.toString())
+        return
+      }
+      console.log('Failed to initialise helia', error)
+      console.timeEnd('IPFS_INIT')
+      return dispatch({ type: 'IPFS_INIT_FAILED', error })
+    }
+  }
+}
+export default bundle
+
+async function initHelia (ipfsApi) {
+
+  const delegatedClient1 = kuboClient(ipfsApi)
+  const delegatedClient2 = kuboClient({
+    protocol: 'https',
+    port: 443,
+    host: 'node3.delegate.ipfs.io'
+  })
+  const blockstore = new MemoryBlockstore()
+  const datastore = new MemoryDatastore()
```

Comment:
We need to make sure persistent datastore is not used if we're in webui/desktop because ideally kubo would be storing those blocks for us. 

Also, we don't really want to instantiate helia in those instances either.. no need to be running two IPFS nodes, but it might not *hurt* for now. open to feedback @hacdias @lidel

Reply:
I'd not worry about persistent data-store:

> increase number of providers of blocks (other explore.ipld.io users)

This  probably needs to wait until libp2p provides [webrtc-browser-to-browser](https://connectivity.libp2p.io/#webrtc?tab=browser-to-browser) transport.

Even then, it is fine to provide them over bitswap if anyone asks, but probably don't want a browser node to announce them on DHT or IPNI.

FYSA ProbeLab found out that having many providers which can't be reached starts impacting content routing (increased latency due to wasting time trying to reach offline peers) -- details in https://github.com/protocol/network-measurements/issues/49

> reduce load on gateways

If we fetch from gateway, immutable blocks are returned with proper `Cache-Control` header,
 so browser cache already stores and resuses them, no need for persistent datastore.

Reply:
makes sense! thanks for the info!